assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/SpecificationFactory.java

```diff
@@ -109,4 +118,49 @@ public static Specification isBetween(String attribute, int min, int max) {
   public static Specification isBetween(String attribute, double min, double max) {
     return (root, query, cb) -> cb.between(root.get(attribute), min, max);
   }
+
+  /**
+   * Returns a list of values from the provided column filter.
+   * Correctly handles assessment LocalDate fields, including dot-notated sub entity fields.
+   *
+   * @param cf The column filter to use
+   * @return A list of values
+   */
+  public static List<Object> getDateAwareValuesFromColumnFilter(ColumnFilter cf) {
+    List<Object> valuesList;
+    try {
+      String[] subEntity = StringUtils.split(cf.getName(), DOT);
+      Field field;
+      if (subEntity.length == 1) {
+        field = Assessment.class.getDeclaredField(subEntity[0]);
+      } else {
+        switch (subEntity[0].toLowerCase()) {
+          case "revalidation":
+            field = Revalidation.class.getDeclaredField(subEntity[1]);
+            break;
+          case "outcome":
+            field = AssessmentOutcome.class.getDeclaredField(subEntity[1]);
+            break;
+          case "detail":
+            field = AssessmentDetail.class.getDeclaredField(subEntity[1]);
+            break;
+          default:
+            throw new NoSuchFieldException(subEntity[0]);
+        }
```

Comment:
Is it worth including the full `cf.getName()` and logging in the catch?  I wouldn't do this, but on the basis of passing `subEntity[0]`, `java.lang.ClassNotFoundException` is probably a better fit

It took me a few min to work out why throw an error when we could return the values from here but I like it!

Reply:
Good call